core/domain/customization_args_util.py

```diff
@@ -123,3 +131,217 @@ def validate_customization_args_and_values(
             # not involved (If they are, can we get sample values for the
             # state context parameters?).
             pass
+
+
+def find_subtitled(
+        customization_arg, ca_spec, conversion_fn,
+        content_id_prefix='custarg'):
+    """Helper function that recursively traverses a customization argument
+    spec to locate any SubtitledHtml or SubtitledUnicode objects, generating
+    a content_id prefix by appending name properties found while traversing,
+    and applying a conversion function to the customization argument value.
+
+    Args:
+        customization_arg: dict. The customization dict to be modified: dict
+            with a single key, 'value', whose corresponding value is the
+            value of the customization arg.
+        ca_spec: dict. The spec dictionary.
+        conversion_fn: function. The function to be used for converting the
+            HTML. It is passed the obj_type, customization argument value,
+            content_id prefix generated from schema, and the customization
+            argument name, if availible.
+        content_id_prefix: string. The content_id prefix generated by
+            appending the 'name' property encountered while traversing the
+            ca_spec.
+    """
+    if 'schema' in ca_spec:
+        schema = ca_spec['schema']
+    else:
+        schema = ca_spec
```

Comment:
This seems odd -- a ca_spec is, technically, not a schema. There might be some confusion of concepts here.

Reply:
Redid the logic here.